include/boutexception.hxx

```diff
@@ -28,6 +28,11 @@ public:
   ~BoutException() override;
 
   const char* what() const noexcept override {
+    if (getenv("BOUT_SHOW_BACKTRACE") != nullptr) {
```

Comment:
This is portable, but should be `std::getenv` from `<cstdlib>`

https://en.cppreference.com/w/cpp/utility/program/getenv